presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergOrcFileWriter.java

```diff
@@ -0,0 +1,259 @@
+            NestedField icebergField = icebergSchema.findField(icebergId);
+            verify(icebergField != null, "Cannot find Iceberg column with ID %s in schema %s", icebergId, icebergSchema);
+            valueCountsBuilder.put(icebergId, fileRowCount);
+            if (orcColumnStats.hasNumberOfValues()) {
+                nullCountsBuilder.put(icebergId, fileRowCount - orcColumnStats.getNumberOfValues());
+            }
+            toIcebergMinMax(orcColumnStats, icebergField.type()).ifPresent(minMax -> {
+                lowerBoundsBuilder.put(icebergId, minMax.getMin());
+                upperBoundsBuilder.put(icebergId, minMax.getMax());
+            });
+        }
+        Map<Integer, Long> valueCounts = valueCountsBuilder.build();
+        Map<Integer, Long> nullCounts = nullCountsBuilder.build();
+        Map<Integer, ByteBuffer> lowerBounds = lowerBoundsBuilder.build();
+        Map<Integer, ByteBuffer> upperBounds = upperBoundsBuilder.build();
+        return new Metrics(
+                fileRowCount,
+                null, // TODO: Add column size accounting to ORC column writers
+                valueCounts.isEmpty() ? null : valueCounts,
+                nullCounts.isEmpty() ? null : nullCounts,
+                lowerBounds.isEmpty() ? null : lowerBounds,
+                upperBounds.isEmpty() ? null : upperBounds);
+    }
+
+    private static Set<Integer> getExcludedColumns(List<OrcType> orcRowTypes)
+    {
+        ImmutableSet.Builder<Integer> excludedColumns = ImmutableSet.builder();
+        populateExcludedColumns(orcRowTypes, 0, false, excludedColumns);
+        return excludedColumns.build();
+    }
+
+    private static void populateExcludedColumns(List<OrcType> orcRowTypes, int orcColumnId, boolean exclude, ImmutableSet.Builder<Integer> excludedColumns)
+    {
+        if (exclude) {
+            excludedColumns.add(orcColumnId);
+        }
+        OrcType orcColumn = orcRowTypes.get(orcColumnId);
+        switch (orcColumn.getOrcTypeKind()) {
+            case LIST:
+            case MAP:
+                for (Integer child : orcColumn.getFieldTypeIndexes()) {
+                    populateExcludedColumns(orcRowTypes, child, true, excludedColumns);
+                }
+                return;
+            case STRUCT:
+                for (Integer child : orcColumn.getFieldTypeIndexes()) {
+                    populateExcludedColumns(orcRowTypes, child, exclude, excludedColumns);
+                }
+                return;
+        }
+    }
+
+    private static int getIcebergId(OrcType orcColumn)
+    {
+        String icebergId = orcColumn.getAttributes().get(ORC_ICEBERG_ID_KEY);
+        verify(icebergId != null, "ORC column %s doesn't have an associated Iceberg ID", orcColumn);
+        return Integer.parseInt(icebergId);
+    }
+
+    private static Optional<IcebergMinMax> toIcebergMinMax(ColumnStatistics orcColumnStats, org.apache.iceberg.types.Type icebergType)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 180-192, lines 193-205, lines 206-214, lines 215-223, lines 224-234 into separate methods.